src/flp/types.rs

```diff
@@ -980,13 +1035,26 @@ mod tests {
             .unwrap();
         }
 
+        let len = 23;
+        let sum_vec = f(2, len).unwrap();
+        flp_validity_test(
+            &sum_vec,
+            &vec![nine; 2 * len],
```

Comment:
9 cannot be encoded with 2 bits. Maybe this should be fixed to be independent of the order of checks.